### ISSUE TITLE
Function not returning a missing result if any of the inputs are missing

### ISSUE THREAD

**Issue by akhnatal** [jpmml/jpmml-evaluator#192]
I am trying to use the `Apply#mapMissingTo`  attribute to skip the function and return a missing value.
I would like to apply this behaviour:

> Unless otherwise specified, a function will return a missing result if any of the inputs are missing.

[http://dmg.org/pmml/v4-4/Functions.html](http://dmg.org/pmml/v4-4/Functions.html)

But what ever I assigned to mapMissingTo ("" or "NA" or "null" or "NaN"), it always return a string.
I need missing Values as input for my XGBoost model and the expected behaviour works perfectly in python with sklearn2pmml.

If I use `Apply#mapMissingTo` I've got this error :
`org.jpmml.evaluator.EvaluationException: No PMML data type for the intersection of PMML data types double and string`

If I don't use `Apply#mapMissing` I've got this error :
`org.jpmml.evaluator.FunctionException: Missing value at position 0`

```
<MiningSchema>
	<MiningField name="label" usageType="target"/>
	<MiningField name="driver_lat"/>
	<MiningField name="driver_lon"/>
	<MiningField name="restaurant_lat"/>
	<MiningField name="restaurant_lon"/>
	<MiningField name="customer_lat"/>
	<MiningField name="customer_lon"/>
	<MiningField name="driver_order_restaurant_lat" missingValueTreatment="asValue"/>
	<MiningField name="driver_order_restaurant_lon" missingValueTreatment="asValue"/>
	<MiningField name="driver_order_customer_lat" missingValueTreatment="asValue"/>
	<MiningField name="driver_order_customer_lon" missingValueTreatment="asValue"/>
	<MiningField name="restaurant_id" missingValueTreatment="asValue" invalidValueTreatment="asIs"/>
	<MiningField name="driver_order_restaurant_id" missingValueTreatment="asValue" invalidValueTreatment="asIs"/>
</MiningSchema>
```

```
<LocalTransformations>
	<DerivedField name="eval(X[0] == X[1])" optype="continuous" dataType="double">
		<Apply function="equal" mapMissingTo="">
			<FieldRef field="restaurant_id"/>
			<FieldRef field="driver_order_restaurant_id"/>
		</Apply>
	</DerivedField>
	<DerivedField name="['restaurant_lat', 'restaurant_lon', 'driver_order_restaurant_lat', 'driver_order_restaurant_lon']" optype="continuous" dataType="double">
		<Apply function="x-sin">
			<Apply function="/" >
				<Apply function="*" >
					<Constant dataType="double">0.017453292519943295</Constant>
					<Apply function="-"  mapMissingTo="">
						<FieldRef field="driver_order_restaurant_lat"/>
						<FieldRef field="restaurant_lat"/>
					</Apply>
				</Apply>
				<Constant dataType="double">2.0</Constant>
			</Apply>
		</Apply>
	</DerivedField>
</LocalTransformations>
```

And is it possible to skip the nested `Apply#function` with the `Apply#mapMissingTo`  attribute ?

**Comment by vruusmann** [comment]
The markup `<Apply function="equal" mapMissingTo="">` is interpreted according to PMML specification as "if any of the inputs are missing, return an empty string `""` as a result".

If you want to return a missing value, then simply don't specify the `Apply@mapMissingTo` attribute at all (and there's no point in trying to try string values `"NA"`, `"N/A"` etc, because they are interpreted as string literals).

> org.jpmml.evaluator.EvaluationException: No PMML data type for the intersection of PMML data types double and string

You cannot use an empty string as an operand in arithmetic operations. For example, the expression `2.0 - ""` does not make sense.

> org.jpmml.evaluator.FunctionException: Missing value at position 0

When mixing missing value-aware (eg. arithmetic functions) and missing value-unaware expressions (eg. trigonometric functions), then it is your responsibility to manually guard/performs missingness checks.

In Java (pseudo-)code:
```java
if(x1 != null && x2 != null){
  return Math.sin(x1 + x2);
} else

{
  return null;
}
```

When using the `sklearn2pmml.preprocessing.ExpressionTransformer` transformer class (in Scikit-Learn pipeline), then you can perform missingness checks using `pandas.isnull` and `pandas.notnull` functions.

TLDR: Everything works as laid out in the PMML specification.

**Comment by vruusmann** [comment]
> .. you can perform missingness checks using pandas.isnull and pandas.notnull functions

Of use explicit if statements like `if myvar is None` and/or `if myvar is not None`.

**Comment by akhnatal** [comment]
OK, thanks.
Wrapping with `if pandas.notnull()` works pretty well :
`ExpressionTransformer("(X[0] == X[1]) if pandas.notnull(X[1]) else X[1]")`
